From 7fcc6f8cdb64c317b9b610d13752cee8a76eabd9 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 2 Mar 2021 08:07:19 +0100
Subject: [PATCH 1/1] efi_loader: correct uboot_bin_env.its file format

Up to now the EFI capsule Python tests were always skipped. The reason is
that mkimage fails with:

uboot_bin_env.its:13.21-23.5: Warning (unit_address_vs_reg):
/images/u-boot-bin@100000: node has a unit name, but no reg property
uboot_bin_env.its:24.21-34.5: Warning (unit_address_vs_reg):
/images/u-boot-env@150000: node has a unit name, but no reg property

If a unit in a device-tree has an address, a reg property must be provided.
But adding a reg property is not the solution here.

Since 2017 unit addresses are disallowed for FIT,
cf. common/image-fit.c:1624.

So remove the unit addresses in uboot_bin_env.its.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 test/py/tests/test_efi_capsule/uboot_bin_env.its | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/py/tests/test_efi_capsule/uboot_bin_env.its b/test/py/tests/test_efi_capsule/uboot_bin_env.its
index 31e2f8049f..fc65907481 100644
--- a/test/py/tests/test_efi_capsule/uboot_bin_env.its
+++ b/test/py/tests/test_efi_capsule/uboot_bin_env.its
@@ -10,7 +10,7 @@
 	#address-cells = <2>;
 
 	images {
-		u-boot-bin@100000 {
+		u-boot-bin {
 			description = "U-Boot binary on SPI Flash";
 			data = /incbin/("BINFILE1");
 			compression = "none";
@@ -21,7 +21,7 @@
 				algo = "sha1";
 			};
 		};
-		u-boot-env@150000 {
+		u-boot-env {
 			description = "U-Boot environment on SPI Flash";
 			data = /incbin/("BINFILE2");
 			compression = "none";
-- 
2.30.1

